### PR TITLE
Use fundanmental-mode to open minified file

### PR DESCRIPTION
**Pull request by ztlevi**
This commit aims to solve the freezing problem when Emacs editing minified files. It's very simple by just checking the first line's length.

And it's disabled for `text-mode` files.

### CODE REVIEW

File: layers/+spacemacs/spacemacs-defaults/funcs.el [hunk 1]
```diff
@@ -436,6 +436,23 @@ FILENAME is deleted using `spacemacs/delete-file' function.."
...
+        (- (line-end-position)
+           (line-beginning-position)))))
+
+;; check if the file has been minified
+;; disable for text-mode files
+(defun spacemacs/check-minified-file ()
+  (and
+   (not (member (file-name-extension (buffer-file-name))
+                '("org" "md" "markdown" "txt" "rtf")))
+   (> (spacemacs/get-nth-line-length 1) 500)))
```
Comment (syl20bnr) [inline layers/+spacemacs/spacemacs-defaults/funcs.el]: Can we up this number a bit? 500 seems low to me.

Comment (ztlevi) [reply layers/+spacemacs/spacemacs-defaults/funcs.el]: How much do you want?

**Comment by ztlevi** [comment]
I made an update of the function see [this line](https://github.com/syl20bnr/spacemacs/pull/10982/commits/7cdce32fde9f4f82579bdf398ba791e7dde4ed33#diff-8a41e260024d225e124323deab81aaf3R455). But I'm not sure how can we break the loop as soon as we find a long line and just return `t`.

**Comment by syl20bnr** [comment]
Performance wise this new function is still fine ? Do you know the complexity of the function `line-end-position` ?

**Comment by ztlevi** [comment]
Hi, I pushed a new commit and the modified function checks the first 20 lines. This prevents minified files with comments in the top get passed.

For the question, it's a built-in function. See:

```
line-end-position is a built-in function in ‘src/editfns.c’.

(line-end-position &optional N)

Return the character position of the last character on the current line.
With argument N not nil or 1, move forward N - 1 lines first.
If scan reaches end of buffer, return that position.

This function ignores text display directionality; it returns the
position of the last character in logical order, i.e. the largest
character position on the line.

This function constrains the returned position to the current field
unless that would be on a different line than the original,
unconstrained result.  If N is nil or 1, and a rear-sticky field ends
at point, the scan stops as soon as it starts.  To ignore field
boundaries bind ‘inhibit-field-text-motion’ to t.

This function does not move point.
```

As for the performance, I think if we put the collector in the `spacemacs/get-nth-line-length`, and return collected lines length, it should be better.

**Pull request closed without merging**